Adds std_ddof keyword option to sigma_clipped_stats

larrybradley opened the pull request:
Addresses #6066.

crawfordsm commented:
Thanks @larrybradley -- this seems pretty straight forward and as the default value does not break backward compatibility.  Two minor comments to consider:  1) You could change the assert on the stddev to the exact value knowing the value of N and ddof.  2) Considering the amount of detailed description in the original issue, it might be useful to include the number for that in the changelog as well.  

I don't see either as critical and happy to see this merged once the tests pass.

pllim commented:
I added "what's new needed tag" but feel free to remove it if it is not really needed.

Need to resolve change log conflict.

larrybradley commented:
@crawfordsm I updated the test to compare to actual the std value and added the original issue number to the changelog.  Thanks.

@pllim I added an entry in the "What's New" and rebased.  This is a minor addition, but it can't hurt to include it.  Thanks.

pllim commented:
How does the "what's new" needed tag work? If what's new is already updated in this PR, then can we remove the tag?

pllim commented:
@larrybradley , I don't see any changes to the what's new doc in the diff.

larrybradley commented:
Thanks, @pllim.  It's there now.

pllim commented on docs/whatsnew/2.0.rst:
```diff
@@ -15,7 +15,8 @@ the 1.3.x series of releases.
 
 * `Easier use of efficient bytestring Table columns in Python 3`_
-*
+* A new ``std_ddof`` keyword option was added to
```
This entry appears to be in the wrong section? You need to add your own section title.

larrybradley replied:
So picky! 😃  Based on previous releases, it actually goes at the top *and* has its own section.  I added the section.

pllim commented on docs/whatsnew/2.0.rst:
```diff
@@ -15,7 +15,8 @@ the 1.3.x series of releases.
 
 * `Easier use of efficient bytestring Table columns in Python 3`_
-*
+* A :ref:`new "std_ddof" keyword <whatsnew-2.0-ddof>` option was added
```
You didn't delete the erroneous what's new entry. :sweat_smile:

pllim replied:
Oh wait... I misread the underline. Sorry.

pllim commented:
LGTM. Thanks!

bsipocz commented:
@pllim - I think it's OK to leave the tags in place, so be can have a quick double check before releasing that everything is in place, etc.
